### DISCUSSION TITLE
Hotfix performance issue of frequent API call

### DISCUSSION THREAD

**Discussion by jmaas-onoffice** [onOffice-Web-Org/oo-wp-plugin#338] [_Discarded pitches]
## Plugin calls API on every page load, which is slow when the API translates
The plugin sends an API call on every page load. (https://github.com/onOffice-Web-Org/oo-wp-plugin/blob/master/plugin/Controller/AdminViewController.php#L388) When the API needs to translate the result, it gets really slow (see T#3264915). This means, when my WordPress is set to English, but the enterprise is German by default, all my page loads get really slow, even when I'm not on a page from the plugin.

## As a hotfix, use a faster API call
As a hotfix, we should use a different API call which is faster because it does not need to be translated.

Jakob, who knows both the plugin and the API well, recommended us to use https://apidoc.onoffice.de/actions/informationen-abfragen/estate-categories/:

``` 
{
        "actionid": "urn:onoffice-de-ns:smart:2.5:smartml:action:get",
        "resourceid": "",
        "identifier": "",
        "resourcetype": "estateCategories",
        "parameters": []
}
```

According to (JJ), requests towards the agent log like https://apidoc.onoffice.de/actions/informationen-abfragen/number-of-log-entries/ are expensive and should be avoided.

## Improvents to the logic should be done later
We could imagine improving the logic to only call the API when it is necessary or be smarter with caching. But we currently do not want to invest the time in this. We only want to hotfix the API call that is used.

**Comment by jmaas-onoffice** [comment]
@dai-eastgate @LongTrong-exe Is this a small change? Can we do this during ramp-up (that is until next Tuesday)?

**Reply by dai-eastgate** [reply]
> @dai-eastgate @LongTrong-exe Is this a small change? Can we do this during ramp-up (that is until next Tuesday)?

Yes, I will work in during ramp-up.
